### PR TITLE
Support for more types

### PR DESCRIPTION
**Pull request by elonazoulay**
Add support for integer and real types.

### CODE REVIEW

**Comment by elonazoulay** [comment]
Depends on #4724

**Comment by elonazoulay** [comment]
This pull request no longer depends on any other pull requests

File: presto-pinot/src/main/java/io/prestosql/pinot/PinotSegmentPageSource.java [hunk 1]
```diff
@@ -366,8 +371,8 @@ Block getArrayBlock(int rowIndex, int columnIndex)
             case FLOAT_ARRAY:
                 float[] floatArray = currentDataTable.getDataTable().getFloatArray(rowIndex, columnIndex);
                 blockBuilder = elementType.createBlockBuilder(null, floatArray.length);
-                for (double element : floatArray) {
-                    elementType.writeDouble(blockBuilder, element);
+                for (float element : floatArray) {
+                    blockBuilder.writeInt(floatToIntBits(element)).closeEntry();
```
Comment (martint) [inline presto-pinot/src/main/java/io/prestosql/pinot/PinotSegmentPageSource.java:373]: Why `.closeEntry()`?

File: presto-pinot/src/main/java/io/prestosql/pinot/decoders/RealDecoder.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import static java.lang.Float.floatToIntBits;
+
+public class RealDecoder
+        implements Decoder
+{
+    @Override
+    public void decode(Supplier<Object> getter, BlockBuilder output)
+    {
+        output.writeInt(floatToIntBits(((Number) getter.get()).floatValue())).closeEntry();
```
Comment (martint) [inline presto-pinot/src/main/java/io/prestosql/pinot/decoders/RealDecoder.java:28]: Why `.closeEntry()`

**Pull request merged**